src/plugins/data/common/opensearch_query/filters/geo_shape_filter.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoShapeRelation } from '@opensearch-project/opensearch/api/types';
+import { Filter, FilterMeta } from './meta_filter';
+
+export type Position = number[];
+
+export interface PreIndexedShapeFilter {
+  index: string;
+  id: string;
+  path: string;
+  routing?: string;
+}
+
+export interface Polygon {
+  type: 'Polygon';
+  coordinates: Position[][];
+}
+
+export interface MultiPolygon {
+  type: 'MultiPolygon';
+  coordinates: Position[][][];
+}
+
+// Will support polygon & multi polygon at this moment
+export type ShapeFilter = Polygon | MultiPolygon;
+
+export type GeoShapeFilterMeta = FilterMeta & {
+  params: {
+    shape?: ShapeFilter;
+    indexed_shape?: PreIndexedShapeFilter;
+    relation?: GeoShapeRelation;
+  };
+};
+
+export type GeoShapeFilter = Filter & {
+  meta: GeoShapeFilterMeta;
+  geo_shape: any;
+};
+
+export const isGeoShapeFilter = (filter: any): filter is GeoShapeFilter =>
+  filter && filter.geo_shape;
+
+export const getGeoShapeFilterField = (filter: GeoShapeFilter) => {
+  return filter.geo_shape && Object.keys(filter.geo_shape).find((key) => key !== 'ignore_unmapped');
```

Comment:
This would be safer with the extra check
```suggestion
  return filter?.geo_shape && Object.keys(filter.geo_shape).find((key) => key !== 'ignore_unmapped');
```

Reply:
Here geo_shape is not an optional field for GeoShapeFilter, adding chaining is not required.

Reply:
updated